third-party/rsutils/src/network-adapter-watcher.cpp

```diff
@@ -196,9 +196,12 @@ class network_adapter_watcher_singleton
 #else
         if( _th.joinable() )
         {
-            if( write( _done, &_done, sizeof( _done ) ) != sizeof( _done ) )
-                /* to avoid compiler warning about not using return value */;
-            _th.join();
+            uint64_t incr = 1;  // must be 8-byte integer value
+            auto rv = write( _done, &incr, sizeof( incr ) );
+            if( rv != sizeof( incr ) )
+                LOG_WARNING( "failed to write to network adapter watcher done event: " << rv );
+            else
+                _th.join();
```

Comment:
If we don't join, it will cause program abnormal termination. Maybe retry to signal?

Reply:
This is just to avoid a hang if the write fails (it failed and hung); it's not expected once done right (and now it is)

Reply:
OK, if write fails it's probably an OS error and it means we won't close gracefully. I think it's acceptable.